internal/pkg/warewulfd/api/power.go

```diff
@@ -0,0 +1,200 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/swaggest/usecase"
+	"github.com/swaggest/usecase/status"
+	"github.com/warewulf/warewulf/internal/pkg/bmc"
+	"github.com/warewulf/warewulf/internal/pkg/node"
+	"github.com/warewulf/warewulf/internal/pkg/wwlog"
+)
+
+// PowerResult is the result structure for single node power status operations. get/set power status
+type PowerResult struct {
+	NodeId string `yaml:"node_id" json:"node_id" description:"ID of the node for which power status is being reported"`
+	BmcIp  string `yaml:"bmc_ip" json:"bmc_ip" description:"BMC IP address of the node"`
+	Result string `yaml:"result" json:"result" description:"Power status of the node, e.g., 'on', 'off', 'unknown'"`
+	Err    error  `yaml:"error,omitempty" json:"error,omitempty" description:"Error message if any error occurred while getting power status"`
```

Comment:
Rather than returning an Error in the result, just actually return an HTTP error when there is an error.